include/miral/miral/simulated_secondary_click.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIRAL_SIMULATED_SECONDARY_CLICK_H
+#define MIRAL_SIMULATED_SECONDARY_CLICK_H
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+namespace mir
+{
+class Server;
+}
+
+namespace miral
+{
+/// Enables configuring simulated secondary click at runtime.
+/// \remark Since MirAL 5.3
+class SimulatedSecondaryClick
+{
+public:
+    /// \note `--enable-simulated-secondary-click` has higher precedence than
+    /// [enabled_by_default]
+    explicit SimulatedSecondaryClick(bool enabled_by_default);
+
+    void operator()(mir::Server& server);
+
+    /// Enables simulated secondary click.
+    /// When already enabled, further calls have no effect.
+    SimulatedSecondaryClick& enable();
+
+    /// Disables simulated secondary click.
+    /// When already disabled, further calls have no effect.
+    SimulatedSecondaryClick& disable();
+
+    /// Configures the duration users have to keep the left mouse button down
+    /// to dispatch a secondary click.
+    SimulatedSecondaryClick& hold_duration(std::chrono::milliseconds hold_duration);
```

Comment:
[non-blocking thought] We could avoid the need for armhf symbols by taking a raw int. (That's consistent with, for example, `displacement_threshold()`)